### DISCUSSION TITLE
Using State-Machines

### DISCUSSION THREAD

**Discussion by om327** [FreeOpcUa/opcua-asyncio#800] [General]
Hello, Does anyone have a very simple state-machine that they have used instaed of if or while statements? I've come from a LabVIEW background where these are used quit a lot but seems quite complicated in python? 

Essentailly i have a devices that needs to be polled at different time intervals for different sensors - some are refrehsed every second while others are every 10 seconds, additonally when a process is underway (loading or injecting) you have to poll the device to see if its finished performing the task before you can send another task request. How would this work in an OPC UA server and avoid blocking?

Would a state-machine upset how async works?

**Comment by AndreasHeine** [comment]
if you came from LabView your statemachine is nothing else as e "case" in "whileloop"!
in python there is no "case" but if "condition1": ... elif "condition2": ... else ... 
similar to this:
https://github.com/AndreasHeine/opcua-sub-to-mqtt/blob/master/opcua_client_mqtt_publisher.py#L128

**Reply by harriv** [reply]
In Python 3.10 there's "structural pattern matching", which is kind of case statement: https://www.python.org/dev/peps/pep-0634/
